client/src/components/mergeReportsDialog/mergeReportsDialog.js

```diff
@@ -0,0 +1,196 @@
+            })}
+          </ul>
+        </div>
+      );
+    }
+
+    return null;
+  }
+
+  renderInvalidScans() {
+    const invalidScans = this.getInvalidScans();
+
+    if (_.size(invalidScans)) {
+      return (
+        <div>
+          <span>Failed scans that cannot be included in the merged report:</span>
+          <ul>
+            {invalidScans.map(scan => {
+              return <li key={scan.id}>{scan.name}</li>;
+            })}
+          </ul>
+        </div>
+      );
+    }
+
+    return null;
+  }
+
+  renderButtons() {
+    const { onClose } = this.props;
+    const validCount = _.size(this.getValidScans());
+
+    if (validCount === 0) {
+      return (
+        <Button bsStyle="primary" className="btn-cancel" onClick={onClose}>
+          Close
+        </Button>
+      );
+    }
+
+    return (
+      <React.Fragment>
+        <Button bsStyle="default" className="btn-cancel" onClick={onClose}>
+          Cancel
+        </Button>
+        <Button bsStyle="primary" type="submit" onClick={this.mergeScanResults}>
+          Merge
+        </Button>
+      </React.Fragment>
+    );
+  }
+
+  render() {
+    const { show, scans, onClose } = this.props;
+
+    if (!scans || scans.length === 0 || !scans[0]) {
+      return null;
+    }
+    const validCount = _.size(this.getValidScans());
+    const invalidCount = _.size(this.getInvalidScans());
+
+    let icon;
+    let heading;
+    if (validCount < 2) {
+      icon = <Icon type="pf" name="error-circle-o" />;
+      heading = 'This action is invalid. You must select at least two scans with successful most recent scans.';
+    } else if (invalidCount > 0) {
+      icon = <Icon type="pf" name="warning-triangle-o" />;
+      heading = 'Warning, only the selected scans with successful most recent scans will be included in the report.';
+    } else {
+      icon = <Icon type="pf" name="info" />;
+    }
+
+    return (
+      <Modal show={show} onHide={onClose}>
+        <Modal.Header>
+          <button className="close" onClick={onClose} aria-hidden="true" aria-label="Close">
+            <Icon type="pf" name="close" />
+          </button>
+          <Modal.Title>Merge Reports</Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          <div className="merge-reports-body">
+            <span className="merge-reports-icon">{icon}</span>
+            <span>
+              <div className="merge-reports-heading">
+                <span>{heading}</span>
+              </div>
+              <div>
+                {this.renderValidScans()}
+                {this.renderInvalidScans()}
+              </div>
+            </span>
+          </div>
+        </Modal.Body>
+        <Modal.Footer>{this.renderButtons()}</Modal.Footer>
+      </Modal>
+    );
+  }
+}
+
+MergeReportsDialog.propTypes = {
+  getMergeScanReults: PropTypes.func,
+  show: PropTypes.bool.isRequired,
+  scans: PropTypes.array,
+  onClose: PropTypes.func
```

Comment:
Note:
Callback for onClose, but there's also a prop for `show`. Seems to make sense at first, but feels off a little since we're using Redux. Might be a good candidate for looking at some clarity between using callbacks vs local state vs redux level

Reply:
What I've been going on thus far is: if it is a common dialog we use redux for application state management but if it is a single dialog, pertinent only to a view, then we include it in the view and use the `show` prop to show/hide the dialog (and thus the close callback). I can change this to use redux if preferred.

Reply:
Let's iterate on this one... 
As for converting to Redux,  the explanation works "view based" vs "app level" 

If the component is maintained as is, thinking it can go the other direction, more stateless. Looks like it would work since it's embracing stateless aspects already. The only rethink might be around how the action is handled, but a callback would work too